orders/views.py

```diff
@@ -1,3 +1,107 @@
+
+            if getattr(user.order_set.last(),'status',None) != before_execute:
+                Order.objects.create(
+                    user=user,
+                    shipping_memo="",
+                    total_price=0,
+                    status=before_execute
+                )
+            order = Order.objects.last()
+
+            if Cart.objects.filter(product=product, product_option=option, order=order).exists():
+                return JsonResponse({'MESSAGE':'CART_ALREADY_EXISTS'}, status=400)
+
+            Cart.objects.create(
+                product=product,
+                quantity=quantity,
+                order=user.order_set.last(),
+                product_option=option
+            )
+            refresh_order(order)
+            return JsonResponse({"MESSAGE":'SUCCESS'},status=200)
+        except json.decoder.JSONDecodeError:
+            return JsonResponse({'MESSAGE': 'JSON_DECODE_ERROR'}, status=400)
+        except KeyError:
+            return JsonResponse({'MESSAGE': 'KEY_ERROR'}, status=400)
+        except TypeError:
+            return JsonResponse({'MESSAGE': 'TYPE_ERROR'}, status=400)
+        except ValueError:
+            return JsonResponse({'MESSAGE': 'VALUE_ERROR'}, status=400)
+        except ProductOption.DoesNotExist:
+            return JsonResponse({'MESSAGE':"PRODUCT_OPTION_DOSENT_EXIST"} ,status=400)
+
+    @query_debugger
+    def patch(self, request):
+        try:
+            data = json.loads(request.body)
+            cart = Cart.objects.select_related('product','order').get(id=data['cart_id'])
+
+            if data.get('quantity'):
+                cart.quantity = data.get('quantity')
+            
+            if data.get('option'):
+                option              = cart.product.productoption_set.get(id=data['option'])
+                cart.product_option = option
+            cart.save()
+
+            refresh_order(cart.order)
+            return JsonResponse({'MESSAGE':"UPDATED"},status=200)
+        except json.decoder.JSONDecodeError:
+            return JsonResponse({'MESSAGE': 'JSON_DECODE_ERROR'}, status=400)
+        except Product.DoesNotExist:
+            return JsonResponse({'MESSAGE':"PRODUCT_DOSENT_EXIST"} ,status=400)
+        except ProductOption.DoesNotExist:
+            return JsonResponse({'MESSAGE':"PRODUCT_OPTION_DOSENT_EXIST"} ,status=400)
+        except KeyError:
+            return JsonResponse({'MESSAGE': 'KEY_ERROR'}, status=400)
+        except TypeError:
+            return JsonResponse({'MESSAGE': 'TYPE_ERROR'}, status=400)
+        except ValueError:
+            return JsonResponse({'MESSAGE': 'VALUE_ERROR'}, status=400)
+
+class OrderView(View):
+    @login_required
+    def post(self, request):
+        try:
+            data                = json.loads(request.body)
+            user                = getattr(request,'user',None)
+            order               = Order.objects.last()
```

Comment:
```suggestion
            order               = Order.objects.get(status=1)
```
주문 가운데 현재 주문이 진행중인, 여전히 active한 상태의 주문은, 한 고객의 전체 주문 가운데 단 하나입니다. (여러 주문이 활성화 되어 있지는 않습니다 :)) 그리고 가장 마지막 주문이 active한 상태가 아닌 경우가 있습니다. 따라서 status자체로 관리해주시는 것이 좋습니다